tests/io/test_crosssection.py

```diff
@@ -0,0 +1,267 @@
+import inspect
+from contextlib import nullcontext as does_not_raise
+from pathlib import Path
+from typing import Any, Callable, List, Union
+
+import pytest
+from pydantic.error_wrappers import ValidationError
+
+from hydrolib.core.io.ini.models import (
+    CircleCrsDef,
+    CrossDefModel,
+    CrossSectionDefinition,
+    RectangleCrsDef,
+    XYZCrsDef,
+    YZCrsDef,
+    ZWCrsDef,
+    ZWRiverCrsDef,
+)
+from hydrolib.core.io.ini.parser import Parser, ParserConfig
```

Comment:
I think with the duplicate behaviour within and between the tested classes, you _could_ have introduced some parameterized tests to avoid repetitive tests and to cover all untested behaviour easily. Just a tip!